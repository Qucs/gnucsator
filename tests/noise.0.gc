load c_include.so
load d_trln.so
load d_noise.so
load s_sparam.so
load mgsim/d_va_noise.so
load mgsim/d_vaflow.so

`include ../include/sources.v
`include ../include/lumped.v

verilog

module main ();
// TODO: ground //
R #(.R(1),.Tc1(0.0),.Tc2(0.0),.Temp(26.85),.Tnom(26.85)) R1 (.p(net1),.n(0));
R #(.R(1),.Tc1(0.0),.Tc2(0.0),.Temp(26.85),.Tnom(26.85)) R1 (.p(net2),.n(0));
Inoise #(.i(1.)) N1 (.p(net1),.n(0));
Inoise #(.i(1.)) N1 (.p(net1),.n(0));
Inoise #(.i(1), .e(2.)) N2 (.p(net2),.n(0));
meas_noise #() n1(net1);
meas_noise #() n2(net2);
endmodule // main

main #() m()

// BUG: need dc.
dc

print ac nv(m.n*)
ac 1e6 2e6 * {sqrt(2)}
