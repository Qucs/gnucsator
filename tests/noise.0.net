# Qucs

R:R1 _net1 gnd R="1" Temp="26.85" Tc1="0.0" Tc2="0.0" Tnom="26.85"
R:R2 _net2 gnd R="1" Temp="26.85" Tc1="0.0" Tc2="0.0" Tnom="26.85"
R:R3 _net3 gnd R="1" Temp="26.85" Tc1="0.0" Tc2="0.0" Tnom="26.85"
Inoise:N0 _net1 gnd i=1
Inoise:N1 _net1 gnd i=1
Inoise:N2 _net2 gnd i=2
Inoise:N3 _net3 gnd i=1 e=2
.AC:AC1 Type="lin" Start="1 MHz" Stop="2 MHz" Points="3" Noise="yes"
