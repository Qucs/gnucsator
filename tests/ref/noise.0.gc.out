T: already installed, replacing
stashing as T:0
tline: already installed, replacing
stashing as tline:0
#          
 0.        
#Freq       nv(m.n1)   nv(m.n2)  
 1.Meg      1.4142     1.u       
 1.4142Meg  1.4142     707.11n   
 2.Meg      1.4142     500.n     

